sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -530,8 +530,14 @@ namespace Azure { namespace Core { namespace Http { namespace _detail {
       Azure::Core::Context const& context,
       Azure::DateTime::duration const& pollDuration)
   {
-    // Before doing any work, check to make sure that the context hasn't already been cancelled.
-    context.ThrowIfCancelled();
+    //
+    // Note that we cannot check for cancellation before calling `initiateAction` because it's
+    // possible that the `initiateAction` call is a call to `WinHttpSendRequest` which establishes the
+    // SendContext.
+    //
+    //// Before doing any work, check to make sure that the context hasn't already been
+    //// cancelled.
+    //context.ThrowIfCancelled();
```

Comment:
So, you've said we can't put `context.ThrowIfCancelled()` before `initiateAction()`. Would it make sense to put it right after (line 552) - in this PR?

Reply:
It might be safe, but honestly this is sufficiently fragile that the 800ms sleep isn't the end of the world. And for C++, cancellation is a best-effort construct anyway. 

And if the request *does* complete within the 800ms, we don't want to cancel it.